packages/nodes-base/nodes/WhatsApp/WhatsApp.node.ts

```diff
@@ -49,6 +65,65 @@ export class WhatsApp implements INodeType {
 			...mediaFields,
 			...messageTypeFields,
 			...mediaTypeFields,
+			...getSendAndWaitProperties([], 'message', undefined, {
+				noButtonStyle: true,
+				defaultApproveLabel: '✓ Approve',
+				defaultDisapproveLabel: '✗ Decline',
+			}).filter((p) => p.name !== 'subject'),
 		],
 	};
+
+	webhook = sendAndWaitWebhook;
+
+	nonRoutingOperations = {
+		message: {
+			async [SEND_AND_WAIT_OPERATION](context: IExecuteFunctions) {
+				const resource = context.getNodeParameter('resource', 0) as string;
+				const operation = context.getNodeParameter('operation', 0) as string;
+				const returnData: INodeExecutionData[] = [];
+
+				if (resource === 'message' && operation === SEND_AND_WAIT_OPERATION) {
```

Comment:
Here we have resource and operation on the key and function name, and also inside the function body. Can we simplify by having this function be specific to the resource + operation pair that's already specified?